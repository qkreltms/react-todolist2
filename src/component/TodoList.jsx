import React from "react";
import Todo from "./Todo.jsx";
import AddTodo from "./AddTodo";
import PropTypes from 'prop-types'

import * as actions from '../actions/todo'
import { connect } from 'react-redux'

const TodoList = props => {
  const state = props;

  const todoList = state.todos.length ? (
    state.todos.map(todo => {
      return <Todo todo={todo} key={todo.id} deleteTodo={state.deleteTodo} />;
    })
  ) : (
    <h3> You have nothing to do. yay!</h3>
  );

  return (
    <div>
      {todoList} <AddTodo addTodo={state.addTodo}/>
    </div>
  );
};

TodoList.propTypes = {
    todos: PropTypes.arrayOf(PropTypes.object),
    deleteTodo: PropTypes.func,
    addTodo: PropTypes.func
}

TodoList.defaultProps = {
  todos: [],
  deleteTodo: () => console.warn('deleteTodo is not defined'),
  addTodo: () => console.warn('addTodo is not defined')
}

const mapStateToProps = (state) => ({
  todos: state.todo.todos
})

const mapDispatchToProps = (dispatch) => ({
  deleteTodo: (id) => {
    dispatch(actions.deleteTodo(id))
  },
  addTodo: (todo) => {
    dispatch(actions.addTodo(todo))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(TodoList)
