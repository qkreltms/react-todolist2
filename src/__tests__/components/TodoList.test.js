import React from 'react';
import TodoList from '../../component/TodoList';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('TodoList', () => {
    test('renders without crashing', () => {
        const component = shallow(<TodoList />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
})
