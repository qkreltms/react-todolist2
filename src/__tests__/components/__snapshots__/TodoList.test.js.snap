// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders without crashing 1`] = `
<Connect(TodoList)
  classes={
    Object {
      "addTodoSection": "Connect-TodoList--addTodoSection-1",
    }
  }
/>
`;
